res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,1802 @@
+/**
+ * Gemini G4V controller script
+ *
+ * For Mixxx 2.0.0+
+ * Written by Javier Vilarroig 2018
+ *
+ **/
+
+/*
+ * Mapping architecture
+ *
+ * Class controller - The full controller
+ *     Contains:
+ *         - decks
+ *         - common controls (not in a deck)
+ *         - common leds (not in a deck)
+ *         - common levels (not in a deck)
+ * Class deck - One deck, including the mixer
+ *     Contains:
+ *         - deck controls
+ *         - deck leds
+ *         - deck levels
+ * Class Connection - One connection between a control and a call back
+ * Class control - One control
+ * Class led - One led
+ * Class level - One level
+ */
+
+print("Loading G4V script");
+
+// Connection Class
+// Manages the events connections from Mixxx engine
+//
+// Properties:
+// controlGroup:        Mixxx group name
+// controlName:            Mixxx control name
+// callbackFunction:    Name of the function to be called when the control is triggered
+//
+// Methods:
+// activate():        Activates the connection between Mixxx and the callback function.
+//                    Connection is not built until the connection is activated.
+// deactivate():    Deactivates the connection between Mixxx and the callback function.
+// refresh():        Refreshes the status of the controller.
+var Connection = function(group, control, callback) {
+    this.groupName = group;
+    this.controlName = control;
+    this.callbackFunction = callback;
+    this.conn = 0;
+
+    if (g4v.debug) {
```

Comment:
do we really need this? `print` statements from controller mappings aren't visible anyway unless you pass `--controllerDebug`.